### PR TITLE
Add Cloud PubSub subscriber in mixer to reload cache when notified

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -142,6 +163,16 @@ func NewStore(
 			cacheData[parts[0]] = parts[1]
 		}
 	}
+	st.cacheData = cacheData
```
Comment (pradh) [inline store/store.go:166]: Should writes to cacheData be synchronized with accesses with a mutex or something?

(Sorry for the delayed response.)

File: store/store.go [hunk 2]
```diff
@@ -175,12 +206,50 @@ func NewStore(
...
+				log.Printf("Load cache data got error %s", err)
+			}
+		})
+		if err != nil {
+			log.Fatalf("Cloud pubsub receive: %v", err)
+		}
+	}()
+
+	// Initial branch cachel load.
+	err = st.LoadBranchCache(ctx, gcsFolder)
```
Comment (pradh) [inline store/store.go:243]: What is gcsFolder at init time - the one set by flag? That could be obsolete. Should we do the "latest_branch_cache_version" file read?

Comment (shifucun) [reply store/store.go:243]: I will replace here by reading the  "latest_branch_cache_version" file when it's ready. This needs to be added in the import controller?

Comment (pradh) [reply store/store.go:243]: Yes.

File: store/store.go (hunk 1 above)
Comment (shifucun) [reply store/store.go]: Use atomic.StorePointer to assign the new cache into store

File: store/cache.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+// NewCache returns a new Cache instance.
+func NewCache() *Cache {
+	cache := Cache{}
+	cache.data = map[string]string{}
+	return &cache
+}
+
+// Read is used for key value look up for Cache.
+func (c *Cache) Read(key string) (string, bool) {
+	c.Lock()
```
Comment (pradh) [inline store/cache.go:37]: RLock()?

File: store/cache.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	defer c.Unlock()
+	value, ok := c.data[key]
```
Comment (pradh) [inline store/cache.go:39]: Any reason to not just do:

  return c.data[key]

Comment (shifucun) [reply store/cache.go:39]: That gives syntax error..

File: store/cache.go (hunk 3 above)
Comment (shifucun) [reply store/cache.go]: Done

**Pull request merged**